testcases/PowerNVDump.py

```diff
@@ -0,0 +1,416 @@
+        if not self.is_fadump_supported():
+            raise OpTestError(
+                "fadump not enabled in the kernel, does system has right firmware!!!")
+        if not self.is_fadump_enabled():
+            raise OpTestError("fadump_enabled is off")
+        self.cv_HOST.host_check_command("kdump")
+        os_level = self.cv_HOST.host_get_OS_Level()
+        self.verify_dump_dt_node()
+        self.verify_fadump_unreg()
+        self.verify_fadump_reg()
+        self.cv_HOST.host_run_command("stty cols 300;stty rows 30")
+        self.cv_HOST.host_enable_kdump_service(os_level)
+        print "======================fadump is supported======================="
+        boot_type = self.kernel_crash()
+        self.verify_dump_dt_node(boot_type)
+        self.verify_dump_file(boot_type)
+
+
+class KernelCrash_OnlyKdumpEnable(PowerNVDump):
+
+    def runTest(self):
+        self.setup_test()
+        if self.is_fadump_supported():
+            raise OpTestError("fadump is enabled, please disable(remove fadump=on \
+                               kernel parameter and re-try")
+        if self.is_fadump_param_enabled():
+            raise OpTestError(
+                "fadump=on added in kernel param, please remove and re-try")
+        self.cv_HOST.host_check_command("kdump")
+        os_level = self.cv_HOST.host_get_OS_Level()
+        self.cv_HOST.host_run_command("stty cols 300;stty rows 30")
+        self.cv_HOST.host_enable_kdump_service(os_level)
+        boot_type = self.kernel_crash()
+        self.verify_dump_file(boot_type)
+
+
+class KernelCrash_DisableAll(PowerNVDump):
+
+    def runTest(self):
+        self.setup_test()
+        if self.is_fadump_param_enabled():
+            raise OpTestError(
+                "fadump=on added in kernel param, please remove and re-try")
+        self.cv_HOST.host_check_command("kdump")
+        os_level = self.cv_HOST.host_get_OS_Level()
+        self.cv_HOST.host_run_command("stty cols 300;stty rows 30")
+        self.cv_HOST.host_disable_kdump_service(os_level)
+        boot_type = self.kernel_crash()
+        if boot_type != BootType.NORMAL:
+            msg = "Invalid boot %d after kernel crash instead of normal boot" % int(
+                boot_type)
+            raise OpTestError(msg)
+
+
+class SkirootKernelCrash(PowerNVDump, unittest.TestCase):
+
+    def setup_test(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+        output = self.c.run_command(PROC_CMDLINE)
+        res = ""
+        found = False
+        update = False
+        for pair in output[0].split(" "):
+            if "xmon" in pair:
+                if pair == "xmon=off":
+                    found = True
+                    continue
+                pair = "xmon=off"
+                update = True
+            res = "%s %s" % (res, pair)
+        if found:
+            return
+        if not update:
+            pair = "xmon=off"
+            res = "%s %s" % (res, pair)
```

Comment:
Can't we simplify above hunk? 

if found 
    return
else
  res = "%s xmon=off" % (res)

Reply:
thats the legacy code which got carried over to new file.. I can rework that..